src/status_im/contexts/wallet/send/input_amount/view.cljs

```diff
@@ -94,91 +102,111 @@
-                                           (not (nil? routes))
-                                           (not loading-suggested-routes?))]
+            on-confirm                                  (or default-on-confirm handle-on-confirm)
+            crypto-decimals                             (or token-decimals default-crypto-decimals)
+            crypto-limit                                (or default-limit-crypto
+                                                            (utils/get-standard-crypto-format
+                                                             token
+                                                             token-balance))
+            fiat-limit                                  (.toFixed (* token-balance conversion-rate) 2)
+            current-limit                               (if @crypto-currency? crypto-limit fiat-limit)
+            valid-input?                                (not
+                                                         (or (string/blank? (controlled-input/input-value
+                                                                             input-state))
+                                                             (<= (controlled-input/numeric-value
+                                                                  input-state)
+                                                                 0)
+                                                             (> (controlled-input/numeric-value
+                                                                 input-state)
+                                                                current-limit)))
+            input-num-value                             (controlled-input/numeric-value input-state)
+            input-amount                                (controlled-input/input-value input-state)
+            confirm-disabled?                           (or (nil? route)
+                                                            (empty? route)
+                                                            (string/blank? input-amount)
+                                                            (<= input-num-value 0)
+                                                            (> input-num-value current-limit))
+            amount                                      (if @crypto-currency?
+                                                          input-amount
+                                                          (number/remove-trailing-zeroes
+                                                           (.toFixed (/ input-amount conversion-rate)
+                                                                     crypto-decimals)))
+            send-amount                                 (rf/sub [:wallet/wallet-send-amount])
+            amount-text                                 (str (.toFixed (js/parseFloat send-amount)
+                                                                       (min token-decimals 6))
```

Comment:
This is the amount string we show to the user. We cap the decimals to 6 to not overflow.